sirepo/package_data/static/js/activait.js

```diff
@@ -1096,59 +1096,140 @@ SIREPO.app.directive('imagePreviewPanel', function(requestSender) {
+                <img class="img-responsive colA{{ method + ($index + 1) }}" />
+                <div data-ng-if="xIsParams"> <br/> <b>{{ parameters[$index] }}</b> </div>
+              </div>
+              <div class="{{ colClass() }}">
+                <img class="img-responsive colB{{ method + ($index + 1) }}" />
+                <div data-ng-if="imageToLabels" class="text-center"> <br/> <b>{{ labels[$index] }}</b> </div>
+              </div>
+              <div data-ng-if="hasThirdColumn" class="col-md-4">
+                <img class="img-responsive pred{{ method + ($index + 1) }}" />
+              </div>
+            </div>
+          </div>
           <div data-ng-if="isLoading()" data-sim-state-progress-bar="" data-sim-state="simState"></div>
           <div data-ng-if="dataFileMissing">Data file {{ fileName }} is missing</div>
           <div data-ng-if="! isLoading() && multiPage">
-            <div class="pull-left">
-              <button class="btn btn-primary" title="first" data-ng-click="first()">|<</button>
+            <div data-ng-if="numPages > 1" class="pull-left">
+              <button class="btn btn-primary" title="first" data-ng-disabled="! canUpdateUri(-1)" data-ng-click="first()">|<</button>
               <button class="btn btn-primary" title="previous" data-ng-disabled="! canUpdateUri(-1)" data-ng-click="prev()"><</button>
             </div>
-            <div class="pull-right">
-                page {{ idx + 1 }} of {{ uris.length }}
+            <div data-ng-if="numPages > 1" class="pull-right">
+                page {{ page() }} of {{ numPages }}
               <button class="btn btn-primary" title="next" data-ng-disabled="! canUpdateUri(1)" data-ng-click="next()">></button>
-              <button class="btn btn-primary" title="last" data-ng-click="last()">>|</button>
+              <button class="btn btn-primary" title="last" data-ng-disabled="! canUpdateUri(1)" data-ng-click="last()">>|</button>
             </div>
           </div>
         </div>
         `,
         controller: function($scope, appState) {
             let loading = true;
-            let numPages = 0;
-            $scope.uris = null;
-            $scope.idx = 0;
+            $scope.numPages = 0;
+            $scope.imagesPerPage = 3;
+            $scope.pageImages = SIREPO.UTILS.indexArray($scope.imagesPerPage);
+            $scope.colA = null;
+            $scope.colB = null;
+            $scope.pred = null;
+            $scope.ColAName = 'Image';
+            $scope.ColBName = 'Contour';
+            $scope.predColName = 'Predicted';
+            $scope.imageIdx = 0;
             $scope.dataFileMissing = false;
-            $scope.canUpdateUri = increment => {
-                return $scope.idx + increment >= 0 && $scope.idx + increment < numPages;
+            $scope.hasThirdColumn = true;
+
+            const pageIndex = () => {
+                return $scope.imageIdx / $scope.imagesPerPage;
+            };
```

Comment:
I don't know where we landed re: "braceless" returns (here `const pageIndex = () => $scope.imageIdx / $scope.imagesPerPage;`) - but I advocate consistency within a directive at least. Though `canUpdateUri()` has a complex enough return value that I think it should use braces. Well, I leave it to you